drafts/k8s-workshop-in-a-box.md

```diff
@@ -0,0 +1,32 @@
+# K8s workshop in a box
+
+We recently organised an internal Kubernetes workshop at our office. Each participant got their own 3-node Kubernetes cluster and all they needed was a laptop with an SSH client, no other software had to be installed! How? Read on to find out!
+
+## Why
+
+I wanted participants to have a real cluster to experiment with, but didn't want to ask everyone to install Minikube. It's heavy weight and I didn't want people to have to mess with this, workshops should have a low as possible barrier to entry.
+
+To pull this off I used [Kind](https://kind.sigs.k8s.io/) (Kubernetes In Docker) + a bit of scripting and Unix magic! 💪
+
+## How it works
+
+It works as follows:
+
+1. A single, beefy cloud server is used to host all Kubernetes clusters.
+2. Every clusters runs 3 Docker containers (via Kind) and is fully isolated from the other clusters. Every cluster consists of one master and two worker nodes.
+3. The clusters are (automatically) provisioned in advance to the workshop, as creating the clusters is quite resource intensive and takes a couple of minutes per cluster.
+4. During the workshop participants can claim their own cluster via SSH and then access their cluster via a new SSH session.
+
+You can see the process from a participant perspective in action below:
+
+[GIF here]
```

Comment:
That's this GIF:

![](https://raw.githubusercontent.com/kabisa/k8s-workshop-in-a-box/master/demo.gif)